common/lib/main.s

```diff
@@ -0,0 +1,20 @@
+.text
+.global __main
+.global __exit
+
+__main:
+    TSX
+    STX __save_sp
+    JMP main
+
+__exit:
```

Comment:
Add a comment describing the mechanics of how this works; in particular the logic of how it uses to the values saved in the `__main` wrapper.